src/main/java/com/google/neighborgood/helper/TaskGroup.java

```diff
@@ -19,101 +19,65 @@
     this.userLoggedIn = userService.isUserLoggedIn();
-    this.userId = this.userLoggedIn ? userService.getCurrentUser().getUserId() : "null";
     this.datastore = DatastoreServiceFactory.getDatastoreService();
-    this.tasks = new StringBuilder();
     this.currentTaskCount = 0;
     this.endOfQuery = false;
+    this.tasks = new ArrayList<>();
   }
 
   /** addTask method adds a single task HTML string to the tasks variable */
   public void addTask(Entity entity) {
 
-    tasks
-        .append("<div class='task' data-key='")
-        .append(KeyFactory.keyToString(entity.getKey()))
-        .append("'>");
-    if (this.userLoggedIn) {
-      tasks.append("<div class='help-overlay'>");
-      tasks.append("<div class='exit-help'><a>&times</a></div>");
-      tasks.append("<a class='confirm-help'>CONFIRM</a>");
-      tasks.append("</div>");
-    }
-    tasks.append("<div class='task-container'>");
-    tasks.append("<div class='task-header'>");
-    tasks.append("<div class='user-nickname'>");
-
-    // Checks if tasks's user nickname has already been retrieved,
-    // otherwise retrieves it and temporarily stores it
-    String taskOwner = (String) entity.getProperty("Owner");
-    if (this.usersNicknames.containsKey(taskOwner)) {
-      tasks.append(this.usersNicknames.get(taskOwner));
+    String taskOwnerId = (String) entity.getProperty("Owner");
+    String taskOwnerNickname = null;
+    Double taskLat = null;
+    Double taskLng = null;
+    if (this.usersInfo.containsKey(taskOwnerId)) {
+      taskOwnerNickname = this.usersInfo.get(taskOwnerId).getUserNickname();
+      taskLat = this.usersInfo.get(taskOwnerId).getUserLat();
+      taskLng = this.usersInfo.get(taskOwnerId).getUserLng();
     } else {
       Key taskOwnerKey = entity.getParent();
       try {
         Entity userEntity = this.datastore.get(taskOwnerKey);
-        String userNickname = (String) userEntity.getProperty("nickname");
-        this.usersNicknames.put(taskOwner, userNickname);
-        tasks.append(userNickname);
+        User taskUser = new User(userEntity);
+        taskOwnerNickname = taskUser.getUserNickname();
+        taskLat = taskUser.getUserLat();
+        taskLng = taskUser.getUserLng();
+        this.usersInfo.put(taskOwnerId, taskUser);
       } catch (EntityNotFoundException e) {
         System.err.println(
             "Unable to find the task's owner info to retrieve the owner's nickname. Setting a default nickname.");
-        tasks.append("Neighbor");
-      }
-    }
-    tasks.append("</div>");
-    if (this.userLoggedIn) {
-      // changes the Help Button div if the current user is the owner of the task
-      if (!this.userId.equals(taskOwner)) {
-        tasks.append("<div class='help-out'>HELP OUT</div>");
-      } else {
-        tasks.append(
-            "<div class='help-out disable-help' title='This is your own task'>HELP OUT</div>");
+        taskOwnerNickname = "Neighbor";
```

Comment:
nit: How about changing the nickname to "Your Friendly Neighbor"? 😄

Reply:
I like that idea! 😃